File: 0071-simplify-path/0071-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        path = path.split("/")
        s = []
        for i in path:
            if i == "..":
                if s:
                    s.pop(-1)
            elif i == "." or i == "":
                continue
            else:
                s.append(i)
        return "/" + "/".join(s)

File: 0071-simplify-path/0071-simplify-path.rb
# @param {String} path
# @return {String}
def simplify_path(path)
    path = path.split("/")
    s = []
    for i in 0..path.length()-1
        if path[i] == ".."
            if s
                s.pop()
            end
        elsif path[i] == "." or path[i] == ""
            next
        else
            s << path[i]
        end
    end
    return "/" + s.join("/")
end